lessons/2_more_strings_and_functions/lesson.py

```diff
@@ -372,25 +419,85 @@ def inner_func():
 # 1. Write a function that takes a string, sorts the characters in the string, and returns a string
 #    with the individual characters sorted. Name this function sort_chars.
 
-# 2. Write a function that takes a string and returns the value of the character at the index which is half of the
-#    total length of the string  if the string is even, the value of the penultimate character if the string is odd, 
+def sort_chars(string):
+    string_sorted = sorted(string)
+    string_sorted = "".join(string_sorted)
+    return string_sorted
+
+print(my_random_chars, "sorted is", sort_chars(my_random_chars))
+
+# # i realized this literally returns e.g. 'the total length of the string', 'l;k2;34lksd;lkfg', 'is', 16
+# # how do i have it return e.g. the total length of the string l;k2;34lksd;lkfg is 16
+
+# # # this seems to be why it is failing test.py
+# # # i think i fixed it, by just pulling the commentary out of the function and into the print.
+
+# 2. Write a function that takes a string and returns the value of the character at the index which is
+#    half of the total length of the string  if the string is even,
+#    the value of the penultimate character if the string is odd, 
 #    and 'Dobby' if an empty-string or length 1. Name this function dobby_search.
 
+def dobby_search(string):
+    if (len(string) % 2) == 0:
+        return string[len(string)//2]
+    else:
+        return string[len(string)-1]
+
+print(dobby_search(my_random_chars))
+
+# # # this one is failing the test, but i am not sure why.
+
 # 3. Write a function that takes in a string and return the ordinal value of the minimum char in the string
 #    or -1 if it is an empty string. Name the function min_to_ord.
 
+def min_to_ord(string):
+    if ord(min(string)) >= 0:
+        return ord(min(string))
+    else:
+        return -1
+
+print(min_to_ord(my_random_chars))
+
+
 # 4. Write a function takes in a string and returns the ordinal value of the maximum char in the string
 # or -1 if it is an empty string. Name the function max_to_ord.
 
-# 5. Write a function that takes in two argumnets: the first a string and the second a character. Return
+def max_to_ord(string):
+    if ord(max(string)) >= 0:
+        return ord(max(string))
+    else:
+        return -1
```

Comment:
def max_to_ord(string):
   if not string:
      return -1
   return ord(max(string))

Reply:
This was my solution, which is a little bit more advanced and I wasn't expecting of you:


```def min_max_helper(string, func):
    return ord(func(string))

def min_to_ord(string):
    if not string:
        return -1
    return min_max_helper(string, min)

def max_to_ord(string):
    if not string:
        return -2
    return min_max_helper(string, max)
```